### ISSUE TITLE
output inconsistency for masked vs compressed model [structured pruning]

### ISSUE THREAD

**Issue by SolomidHero** [microsoft/nni#5095]
**Describe the issue**:

Model compression in NNI breaks numerical consistency when I use it for structured pruned model.

**Environment**:
- NNI version: 2.8
- Client OS: MacOS

**How to reproduce it?**:
```py3
import torch.nn as nn
from nni.compression.pytorch.pruning import FPGMPruner, L1NormPruner, LevelPruner
from nni.compression.pytorch.speedup import ModelSpeedup

m = nn.Sequential(
    nn.Linear(5, 7),
)
_ = torch.nn.init.kaiming_normal_(m[0].weight)
m_input = torch.randn(20, 5)

print(m(m_input).sum())

config_list = [{
    'sparsity_per_layer': 0.5,
    'op_types': ['Linear']
}]
pruner = L1NormPruner(m, config_list)
masked_m, masks = pruner.compress()
pruner._unwrap_model()

print(m(m_input).sum())
print(masked_m(m_input).sum())
ModelSpeedup(m, m_input, masks, 20).speedup_model()

print(m(m_input).sum())
print(masked_m(m_input).sum())
```

example output (see printed lines of models output)
```pytb
tensor(17.3238, grad_fn=<SumBackward0>)  # usual
tensor(13.3754, grad_fn=<SumBackward0>)  # masked
tensor(13.3754, grad_fn=<SumBackward0>)  # masked
[2022-08-29 17:38:18] start to speedup the model
2022-08-29 17:38:18 | INFO | nni.compression.pytorch.speedup.compressor | start to speedup the model
2022-08-29 17:38:18 | INFO | FixMaskConflict | {}
2022-08-29 17:38:18 | WARNING | FixMaskConflict | no multi-dimension masks found.
2022-08-29 17:38:18 | INFO | FixMaskConflict | Dectected conv prune dim" 0
[2022-08-29 17:38:18] infer module masks...
2022-08-29 17:38:18 | INFO | nni.compression.pytorch.speedup.compressor | infer module masks...
[2022-08-29 17:38:18] Update mask for 0
2022-08-29 17:38:18 | INFO | nni.compression.pytorch.speedup.compressor | Update mask for 0
[2022-08-29 17:38:18] Update the indirect sparsity for the 0
2022-08-29 17:38:18 | INFO | nni.compression.pytorch.speedup.compressor | Update the indirect sparsity for the 0
[2022-08-29 17:38:18] resolve the mask conflict
2022-08-29 17:38:18 | INFO | nni.compression.pytorch.speedup.compressor | resolve the mask conflict
[2022-08-29 17:38:18] replace compressed modules...
2022-08-29 17:38:18 | INFO | nni.compression.pytorch.speedup.compressor | replace compressed modules...
[2022-08-29 17:38:18] replace module (name: 0, op_type: Linear)
2022-08-29 17:38:18 | INFO | nni.compression.pytorch.speedup.compressor | replace module (name: 0, op_type: Linear)
[2022-08-29 17:38:18] replace linear with new in_features: 5, out_features: 4
2022-08-29 17:38:18 | INFO | nni.compression.pytorch.speedup.compress_modules | replace linear with new in_features: 5, out_features: 4
[2022-08-29 17:38:18] speedup done
2022-08-29 17:38:18 | INFO | nni.compression.pytorch.speedup.compressor | speedup done
tensor(0.6746, grad_fn=<SumBackward0>)  # compressed
tensor(0.6746, grad_fn=<SumBackward0>)  # compressed
```

**Comment by J-shang** [comment]
nice catch, this because during `pruner.compress()`, `bias_mask` is not generated, we have fixed it in nni 2.9, you could have a try on 2.9alpha.

`pip install --extra-index-url https://test.pypi.org/simple/ nni==2.9a2`

**Comment by ultmaster** [comment]
Closing as the problem has been fixed in 2.9.

If there are further inquiries, please reopen.
